src/update-process-env.js

```diff
@@ -36,6 +33,10 @@ function updateProcessEnv (launchEnv) {
     for (let key in envToAssign) {
       if (!ENVIRONMENT_VARIABLES_TO_PRESERVE.has(key)) {
         process.env[key] = envToAssign[key]
+      } else {
+        if (!process.env[key] && envToAssign[key]) {
```

Comment:
This is related to https://github.com/atom/atom/pull/12562#discussion-diff-78041627R91 but I think this new behavior may be too pessimistic. It would prevent the changing of an existing value in process.env which is not the expected behavior. I will write up a spec to validate this assumption and fix if true.